### PR TITLE
[Docker build] add separate visibility env vars for different db instance

### PR DESCRIPTION
**Pull request by sw-yx**
**What changed?**

add separate visibility env vars for different db instance - some deployment environments, like Render, do not allow for creating another db on the same db instance. There is no hard requirement inside of Temporal, but this assumption is baked into the docker image - so we need to offer separate env vars that people can override.

closes https://github.com/temporalio/temporal/issues/1388

**How did you test it?**

not yet tested

**Potential risks**

n/a

**Is hotfix candidate?**
no

### CODE REVIEW

**Review by yiminc — APPROVED** [review]
I'm assuming that if visibility ones are not defined, it fall back to the primary ones.

File: docker/config_template.yaml [hunk 1]
```diff
@@ -76,10 +80,14 @@ persistence:
...
                 databaseName: "{{ default .Env.VISIBILITY_DBNAME "temporal_visibility" }}"
-                connectAddr: "{{ default .Env.MYSQL_SEEDS "" }}:{{ default .Env.DB_PORT "3306" }}"
+                {{- $vis_seeds := default .Env.VISIBILITY_MYSQL_SEEDS .Env.MYSQL_SEEDS -}}
+                {{- $vis_port := default .Env.VISIBILITY_DB_PORT .Env.DB_PORT -}}
+                connectAddr: "{{ default $vis_seeds "" }}:{{ default $vis_port "3306" }}"
                 connectProtocol: "tcp"
-                user: "{{ default .Env.MYSQL_USER "" }}"
-                password: "{{ default .Env.MYSQL_PWD "" }}"
+                {{- $vis_user := default .Env.VISIBILITY_MYSQL_USER .Env.MYSQL_USER -}}
+                {{- $vis_pwd := default .Env.VISIBILITY_MYSQL_PWD .Env.MYSQL_PWD -}}
```
Comment (alexshtin) [inline docker/config_template.yaml:88]: Move this one line up to match with Cassandra (define all 4 vars first then use them).

File: docker/config_template.yaml [hunk 2]
```diff
@@ -33,11 +33,15 @@ persistence:
                     serverName: {{ default .Env.CASSANDRA_HOST_NAME "" }}
         visibility:
             cassandra:
-                hosts: "{{ default .Env.CASSANDRA_SEEDS "" }}"
+                {{- $vis_seeds := default .Env.VISIBILITY_CASSANDRA_SEEDS .Env.CASSANDRA_SEEDS -}}
```
Comment (alexshtin) [inline docker/config_template.yaml:36]: I don't like abbreviations like this. Can you change `vis` to `visibility` everywhere.

**Review by alexshtin — APPROVED** [review]

**Comment by sw-yx** [comment]
@ yimin if you see this - yes, thats the fallback strategy so it doesnt break any experience for existing users

**Comment by codenation-nl** [comment]
How does this MR relate to the **temporalio/auto-setup** docker container? In this container there is an [auto-setup ](https://github.com/temporalio/docker-builds/blob/main/docker/auto-setup.sh) script which creates the temporal and the temporal_visibility databases+schemas. Am I understanding correctly that this script should be updated to also respect the separate temporal visibility config parameters?

**Pull request merged**